file/lib/src/sentry_file.dart

```diff
@@ -0,0 +1,387 @@
+// Adapted from https://github.com/ueman/sentry-dart-tools/blob/8e41418c0f2c62dc88292cf32a4f22e79112b744/sentry_plus/lib/src/file/sentry_file.dart
+
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+
+typedef Callback<T> = FutureOr<T> Function();
+
+/// The Sentry wrapper for the File IO implementation that creates a span
+/// out of the active transaction in the scope.
+/// The span is started before the operation is executed and finished after.
+/// The File tracing isn't available for Web.
+///
+/// Example:
+///
+/// ```dart
+/// import 'dart:io';
+///
+/// final file = File('test.txt');
+/// final sentryFile = SentryFile(file);
+/// // span starts
+/// await sentryFile.writeAsString('Hello World');
+/// // span finishes
+/// ```
+///
+/// All the copy, create, delete, open, rename, read, and write operations are
+/// supported.
+@experimental
+class SentryFile implements File {
+  SentryFile(
+    this._file, {
+    @internal Hub? hub,
+  }) : _hub = hub ?? HubAdapter() {
+    _hub.options.sdk.addIntegration('SentryFileTracing');
+  }
+
+  final File _file;
+  final Hub _hub;
+
+  @override
+  Future<File> copy(String newPath) {
+    return _wrap(() async => _file.copy(newPath), 'file.copy');
+  }
+
+  @override
+  File copySync(String newPath) {
+    return _wrapSync(() => _file.copySync(newPath), 'file.copy');
+  }
+
+  @override
+  Future<File> create({bool recursive = false}) {
+    return _wrap(
+      () async => _file.create(recursive: recursive),
+      'file.write',
+    );
+  }
+
+  @override
+  void createSync({bool recursive = false, bool exclusive = false}) {
+    return _wrapSync(
+      () => _file.createSync(recursive: recursive),
```

Comment:
```suggestion
      () => _file.createSync(recursive: recursive, exclusive: exclusive),
```

Should we pass also the `exclusive`?

Reply:
Nope, that will come in the next version due to being a breaking change in the Dart language.